lib/get-config.js

```diff
@@ -93,9 +93,29 @@ export async function resolveProjectConfig(workingDir, options) {
   try {
     const { default: config } = await import(configPath);
     return config;
-  } catch {
+  } catch (error) {
     // Fallback to ensure we can load CJS configs in Node versions before 16 (TODO: remove eventually).
-    return require(configPath); // eslint-disable-line import/no-dynamic-require
+    return requireFallback(configPath, error);
+  }
+}
+
+/**
+ * @todo remove this fallback once we drop support for Node 14
+ *
+ * @param {string} requirePath path of module to require
+ * @param {Error} importError error thrown by import()
+ * @throws {Error} error thrown by require() or importError if the module is ESM
+ */
+function requireFallback(requirePath, importError) {
+  try {
+    return require(requirePath); // eslint-disable-line import/no-dynamic-require
+  } catch (error) {
+    if (importError && error?.code === 'ERR_REQUIRE_ESM') {
```

Comment:
`error?.code` - is optional chaining necessary, shouldn't an error always exist?